Add predecode hook

tu-maurice opened the pull request:
This change adds the possibility to let the application using this library to
preprocess the content of the http response. This is especially necessary for
APIs that prefix their answer with garbage to prevent CSRF.

geofffranks commented:
Neat, thanks!

Can you update the readme, pod, and change log for the new feature?

tu-maurice commented:
Will do, however it'll be a few hours before I can work on this though.

tu-maurice committed:
Add predecode hook

This change adds the possibility to let the application using this library to
preprocess the content of the http response. This is especially necessary for
APIs that prefix their answer with garbage, to prevent CSRF.

tu-maurice commented:
there we go.

geofffranks commented:
Should be available on cpan shortly

> On Jun 29, 2019, at 7:24 AM, Maurice <notifications@github.com> wrote:
> 
> there we go.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/geofffranks/json-api/pull/5?email_source=notifications&email_token=ABWZEUVYDZLAFY7FJN4YDJ3P45BAPA5CNFSM4H37U4M2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY3XCIQ#issuecomment-506949922>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABWZEUTKFSPIZIUU3SRBMIDP45BAPANCNFSM4H37U4MQ>.
>
